### PR TITLE
wip snapcraft filename template

### LINKED ISSUE #322: Name template for snap packages
Our "final" pipes all have the option of templating the final name, maybe we should add this feature to snaps as well.

### PR DESCRIPTION
**Pull request by caarlos0**
closes #322

### CODE REVIEW

**Comment by caarlos0** [comment]
cc/ @elopio

File: pipeline/defaults/defaults.go [hunk 1]
```diff
@@ -37,6 +37,12 @@ func (Pipe) Run(ctx *context.Context) error {
 	if ctx.Config.ProjectName == "" {
 		ctx.Config.ProjectName = ctx.Config.Release.GitHub.Name
 	}
+	if ctx.Config.Snapcraft.Name == "" {
+		ctx.Config.Snapcraft.Name = ctx.Config.ProjectName
```
Comment (elopio) [inline pipeline/defaults/defaults.go]: This feels like the wrong place, don't you think?
The snapcraft pipe should take care of all the things snapcraft. This default file seems like a catch all that goes against the abstraction and modularity of releaser.

But well, this was just a comment that nobody asked about possible changes in the design, for the future. If it works now, it is good for me :D

File: pipeline/snapcraft/snapcraft.go [hunk 2]
```diff
@@ -143,10 +141,17 @@ func create(ctx *context.Context, folder, arch string, binaries []context.Binary
...
 	}
 
-	snap := filepath.Join(
-		ctx.Config.Dist,
-		ctx.Config.ProjectName+"_"+metadata.Version+"_"+arch+".snap",
-	)
+	var target = buildtarget.Target{OS: "linux", Arch: arch}
+	if strings.Contains(arch, "arm") {
+		target.Arch = "arm"
+		target.Arm = strings.Replace(arch, "arm", "", 1)
```
Comment (elopio) [inline pipeline/snapcraft/snapcraft.go]: could this transformation be hidden inside ForSnap, instead of a prerequisite step?

**Comment by caarlos0** [comment]
this was the wrong way of fixing it anyways

**Pull request closed without merging**